src/main.cpp

```diff
@@ -1102,7 +1106,9 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
 
         // Check against previous transactions
         // This is done last to help prevent CPU exhaustion denial-of-service attacks.
-        if (!CheckInputs(tx, state, view, 0, true, STANDARD_SCRIPT_VERIFY_FLAGS, true))
+        unsigned int flags = STANDARD_SCRIPT_VERIFY_FLAGS
+                           | (protocol_cleanup? SCRIPT_VERIFY_PROTOCOL_CLEANUP: 0);
```

Comment:
If you want to be cute, `| (-protocol_cleanup & SCRIPT_VERIFY_PROTOCOL_CLEANUP)`. One less `JMP`.
```suggestion
                           | (-protocol_cleanup & SCRIPT_VERIFY_PROTOCOL_CLEANUP);
```

Reply:
Cute trick, but Isn't this implementation defined behavior because signed numbers aren't assumed to be two's complement?

Reply:
Followed this up a little more, and for the record this seems to be well defined, to my surprise. I think I'm going to leave it as-is, however, since the intent is clearer with an explicit conditional.